src/main/httpCommandServer.ts

```diff
@@ -41,18 +52,35 @@ export default class HttpCommandServer {
     console.log('CLI server is now ready.');
   }
 
-  shutdown() {
-    this.server.close();
-  }
-
   protected handleRequest(request: http.IncomingMessage, response: http.ServerResponse) {
-    if (!this.basicAuth(request.headers.authorization ?? '')) {
-      response.writeHead(401, { 'Content-Type': 'text/plain' });
-    } else {
-      response.writeHead(200, { 'Content-Type': 'text/plain' });
-      response.write('Nothing to see here yet.');
+    try {
+      if (!this.basicAuth(request.headers.authorization ?? '')) {
+        response.writeHead(401, { 'Content-Type': 'text/plain' });
+
+        return;
+      }
+      const method = request.method;
+      const url = new URL(request.url as string, `http://${ request.headers.host }`);
+      const path = url.pathname;
+      const pathParts = path.split('/');
+      const commandName = pathParts[0] || pathParts[1];
+      // TODO: Further processing of path parts, query parameters, and -d payload to be done later.
+      const command = this.lookupCommand(method ?? 'GET', commandName);
+
+      if (!command) {
+        response.writeHead(404, { 'Content-Type': 'text/plain' });
+        response.write(`Unknown command: ${ method } ${ commandName }`);
+
+        return;
+      }
+      command(request, response);
```

Comment:
Altenatively, use `command.call(this, request, response)`, so we don't need the extra `bind` on line 115.  Nothing about `lookupCommand` indicates that it returns a bound function.

Reply:
Sometimes you need that 90s-era javascript. Like here.